x/staking/keeper/grpc_query.go

```diff
@@ -108,6 +109,12 @@ func (k Querier) ValidatorDelegations(ctx context.Context, req *types.QueryValid
 		pageRes *query.PageResponse
 	)
 	pageRes, err = query.Paginate(delStore, req.Pagination, func(delAddr, value []byte) error {
+		// Check the store to see if there is a value stored under the key
+		key := store.Get(types.NextMigrateDelegationsByValidatorIndexKey)
+		if key != nil {
+			// Users will never see this error as if there is an error the function defaults to the legacy implementation below
+			return fmt.Errorf("store migration is not finished, try again later")
+		}
```

Comment:
Can we add a TODO to remove after v26, so we don't keep code that people a year from now will not remember why it was added.